crates/api/src/peer_meta_store.rs

```diff
@@ -1,27 +1,48 @@
 use crate::{builder, config, BoxFut, K2Result, Timestamp, Url};
+use bytes::Bytes;
 use futures::future::BoxFuture;
 use std::sync::Arc;
 
 /// A store for peer metadata.
 ///
 /// This is expected to be backed by a key-value store that keys by space, peer URL and key.
 pub trait PeerMetaStore: 'static + Send + Sync + std::fmt::Debug {
-    /// Store a key-value pair for a given space and peer.
+    /// Store a key-value pair for a peer.
     fn put(
         &self,
         peer: Url,
         key: String,
-        value: bytes::Bytes,
+        value: Bytes,
         expiry: Option<Timestamp>,
     ) -> BoxFuture<'_, K2Result<()>>;
 
-    /// Get a value by key for a given space and peer.
+    /// Get a value by key for a peer.
     fn get(
         &self,
         peer: Url,
         key: String,
     ) -> BoxFuture<'_, K2Result<Option<bytes::Bytes>>>;
 
+    /// Mark a peer url unresponsive with an expiration timestamp.
+    //
+    // TODO After the expiry the peer url will be removed from the store.
+    fn mark_peer_unresponsive(
+        &self,
+        peer: Url,
+        expiry: Timestamp,
```

Comment:
Perhaps we could pass an `Arc<AgentInfoSigned>` here so that the caller doesn't have to think about the expiry time? They're going to have to look at up the agent info anyway to work out the expiry, may as well just pass it here for consistency?

Reply:
Another possible motivation for passing it in. I think we might consider storing the signature as the value of this key. That way we can tell apart a new agent info from re-receiving an existing agent info when deciding whether to un-mark. Though maybe we just do that against the peer store and not inside the meta store logic.